woocommerce-services.php

```diff
@@ -1025,6 +1026,77 @@ public function reports_tabs( $reports ) {
 			return $reports;
 		}
 
+		/**
+		 * Send completed order details to Connect Server to check live rates usage.
+		 * Attached to the woocommerce_order_details_after_order_table hook
+		 *
+		 * @param int   $order_id int for completed order ID.
+		 * @param array $posted_data Post data for order.
+		 * @param array $order WC_Order object containing completed order details.
+		 */
+		public function track_completed_order( $order_id, $data, $order ) {
+			$logger = $this->get_logger();
+
+			$packages = WC()->cart->get_shipping_packages();
+			$package  = $packages[0];
+			foreach ( $package['contents'] as $item_key => $item_details ) {
+				ksort( $package['contents'][ $item_key ] );
+			}
+			$package['rates'] = array();
+
+			$shipping_zone    = WC_Shipping_Zones::get_zone_matching_package( $package );
+			$shipping_methods = $shipping_zone->get_shipping_methods( true );
```

Comment:
We should be able simplify this code a little by using a core method.
```suggestion
			$shipping_methods = WC()->shipping()->load_shipping_methods( $package )
```

Reply:
Nice one, just updated this with your suggestion as well.